cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala

```diff
@@ -191,66 +297,269 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
+        """The `EmptyResult` operator eagerly loads all data coming into the `EmptyResult` operator and discards it xxx.""".stripMargin,
       queryText = """CREATE (:Person)""",
       assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("EmptyResult"))
     )
   }
 
+  @Test def produceResult() {
+    profileQuery(
+      title = "Produce Result",
+      text =
+        """The `ProduceResult` operator prepares the result so that it is consumable by the user, such as transforming internal values to user values.
+          |It is present in every single query that returns data to the user, and has little bearing on performance optimisation.""".stripMargin,
+      queryText = """MATCH (n) RETURN n""",
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("ProduceResult"))
+    )
+  }
+
   @Test def nodeByLabelScan() {
     profileQuery(
-      title = "Node by label scan",
-      text = """Using the label index, fetches all nodes with a specific label on them from the node label index.""".stripMargin,
+      title = "Node By Label Scan",
+      text = """The `NodeByLabelScan` operator fetches all nodes with a specific label from the node label index.""".stripMargin,
       queryText = """MATCH (person:Person) RETURN person""",
-      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("LabelScan"))
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("NodeByLabelScan"))
     )
   }
 
   @Test def nodeByIndexSeek() {
     profileQuery(
-      title = "Node index seek",
-      text = """Finds nodes using an index seek. The node variable and the index used is shown in the arguments of the operator.
-                |If the index is a unique index, the operator is called `NodeUniqueIndexSeek` instead.""".stripMargin,
+      title = "Node Index Seek",
+      text =
+        """The `NodeIndexSeek` operator finds nodes using an index seek.
+          |The node variable and the index used is shown in the arguments of the operator.
+          |If the index is a unique index, the operator is instead called <<query-plan-node-unique-index-seek, NodeUniqueIndexSeek>>.""".stripMargin,
       queryText = """MATCH (location:Location {name: 'Malmo'}) RETURN location""",
       assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("NodeIndexSeek"))
     )
   }
 
+  @Test def nodeByUniqueIndexSeek() {
+    profileQuery(
+      title = "Node Unique Index Seek",
+      text = """The `NodeUniqueIndexSeek` operator finds nodes using an index seek within a unique index. The node variable and the index used is shown in the arguments of the operator.
+               |If the index is not unique, the operator is instead called <<query-plan-node-index-seek, NodeIndexSeek>>.""".stripMargin,
+      queryText = """MATCH (t:Team {name: 'Malmo'}) RETURN t""",
```

Comment:
If the index seek is used to solve a MERGE clause, it will also be marked with `(Locking)`. This makes it clear that any nodes returned from the index will be locked, in order to protect from concurrent conflicting updates.

Reply:
👍